### PR TITLE
Add eapis::phys_pci for access to physical PCI devices

### PR DESCRIPTION
**Pull request by cpavlina**
Add a class `eapis::phys_pci` representing physical PCI devices' configuration space with an implementation for x64. Includes integration test to enumerate and dump the system's entire PCI tree.

### CODE REVIEW

File: bfvmm/integration/arch/intel_x64/phys_pci/phys_pci.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+//
```
Comment (connojd) [inline bfvmm/integration/arch/intel_x64/phys_pci/phys_pci.cpp]: Please  move this file to arch/x64 to keep in sync with the source layout

File: bfvmm/include/hve/phys_pci.h [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+     *
+     * @expects bus <= 255, device <= 31, func <= 7, reg <= 0x3C
+     * @ensures
+     *
+     * @param bus PCI bus number
+     * @param device Device number on the bus
+     * @param func Function, for multifunction devices (or 0)
+     * @param reg register index in bytes
+     */
+    static uint8_t read_register_u8(bus_type bus, device_type device,
```
Comment (connojd) [inline bfvmm/include/hve/phys_pci.h]: Please move the static functions to an intrinsics file, similar to lapic_register.{h,cpp}. You could put them in `namespace pci` or something similar

Comment (cpavlina) [reply bfvmm/include/hve/phys_pci.h]: Isn't that what I proposed in the original RFC, that you said not to do? :P

will do.

File: bfvmm/src/hve/arch/x64/phys_pci.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+{
+    const auto ext_bus = static_cast<uint32_t>(bus);
+    const auto ext_dev = static_cast<uint32_t>(device);
+    const auto ext_func = static_cast<uint32_t>(func);
+    const auto ext_reg = static_cast<uint32_t>(reg);
+
+    return ((ext_bus << 16) | (ext_dev << 11) | (ext_func << 8) |
+            (ext_reg & 0xfc) | 0x80000000);
+}
+
```
[the remark this answers is not in the document]
Comment (cpavlina) [reply bfvmm/src/hve/arch/x64/phys_pci.cpp]: Are you sure? PCI is not x64 specific, we should eventually have an aarch64 implementation as well

[the remark this answers is not in the document]
Comment (connojd) [reply bfvmm/src/hve/arch/x64/phys_pci.cpp]: Thats true..so should the phys_pci be moved up into src/hve?

File: bfvmm/integration/arch/intel_x64/phys_pci/phys_pci.cpp (hunk 1 above)
Comment (cpavlina) [reply bfvmm/integration/arch/intel_x64/phys_pci/phys_pci.cpp]: Left this in place as discussed pending moving and fixing `test_all.sh`.

**Comment by connojd** [comment]
Mark the vcpu constructor 'explicit' in the integration test, and make sure all the if statements
are in the form 
```
if() {
;
}
```
I saw one on one line in phys_pci.cpp. I'll need to tweak the astyle config to catch this. 

Last thing, make sure you leave a space after commented code
```
///
code()

///
more_code()

```
Once these changes are done ill merge

**Pull request merged**